src/utils/AsyncEventEmitter.ts

```diff
@@ -1,7 +1,14 @@
 import { Debugger, debug } from 'debug'
-import { StrictEventEmitter, EventMapType } from 'strict-event-emitter'
+import { Emitter } from 'strict-event-emitter'
 import { nextTick } from './nextTick'
 
+export type FunctionEventMap = Record<string, (...args: Array<any>) => any>
+export type FunctionEventMapToStrictEventEmitterMap<
+  Functions extends FunctionEventMap
+> = {
+  [K in keyof Functions]: Parameters<Functions[K]>
```

Comment:
`EventMapType` and `EventMap` types are different across strict-event-emitter 0.2 and 0.5. Adding this transformer type to account for that difference since the events contract in Interceptors uses `Record<string, Function>` index signature to allow specifying the expected return type of the events (since those can be awaited, etc).